analytics/javaspark/src/main/java/mil/nga/giat/geowave/analytic/javaspark/kmeans/KMeansRunner.java

```diff
@@ -102,14 +107,50 @@ public void run()
 		queryOptions.setAdapter(featureAdapterIds);
 
 		// This is required due to some funkiness in GeoWaveInputFormat
-		queryOptions.getAdaptersArray(inputDataStore.createAdapterStore());
+		AdapterStore adapterStore = inputDataStore.createAdapterStore();
+		queryOptions.getAdaptersArray(adapterStore);
```

Comment:
really? thats not good if its true

Reply:
That's an old comment. I have no idea why it's there. I think I split it out just to get a ref to the adapterStore for reuse farther down.